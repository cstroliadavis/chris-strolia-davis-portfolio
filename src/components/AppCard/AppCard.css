.app-card {
    background-color: #98c4c5;
    filter: drop-shadow(.5rem .5rem 3px #5d5d5d);
    margin: 1rem;
    outline: thin solid black;
    padding: 1rem;
    position: relative;
    width: 100%;
}

.app-card .app-ref img {
    aspect-ratio: 19 / 9;
    max-width: 100%;
    object-fit: cover;
}

.app-card .repo {
    margin-top: .5rem;
}

.app-card .repo a {
    background-color: white;
    color: black;
    display: block;
    outline: thin solid black;
    padding: .5rem;
    text-decoration: none;
}

.app-card > .app-description {
    background-color: white;
    display: none;
    filter: drop-shadow(.5rem .5rem 3px #343434);
    left: 4rem;
    margin: -5px;
    max-height: inherit;
    min-height: 50%;
    opacity: .75;
    outline: thin solid black;
    position: absolute;
    top: 4rem;
    width: 75%;
    z-index: 1;
}

.app-card > .app-description > a {
    bottom: 0;
    color: black;
    left: 0;
    padding: 25px 5px 5px;
    position: absolute;
    right: 0;
    text-decoration: none;
    top: 0;
}

.app-card:hover > .app-description {
    display: block;
}
