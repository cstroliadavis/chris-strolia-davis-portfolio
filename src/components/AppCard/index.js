import './AppCard.css';

export default function AppCard(props) {
  const {
    description,
    github,
    image,
    title,
    url,
  } = props;

  return (
    <div className="app-card">
      <h1 className="title">{ title }</h1>
      <div className="app-ref">
        <a href={ url } target="_blank"><img src={ image } alt={ title }/></a>
      </div>
      <div className="repo"><a href={ github } target="_blank">See the code</a></div>
      <div className="app-description"><a href={ url } target="_blank">{ description }</a></div>
    </div>
  );
}
