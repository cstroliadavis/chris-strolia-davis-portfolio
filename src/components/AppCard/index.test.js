import { render, cleanup } from '@testing-library/react';
import AppCard from './index'

afterEach(cleanup);

const mockApp = Object.freeze({
  id: 1,
  title: 'A Title',
  description: 'One of the things I worked on',
  url: 'https://location.app.com',
  github: 'https://github.com/location/app',
  image: 'https://image.app.com/some-image.jpg',
});

describe('AppCard', () => {
  it('renders', () => {
    render(<AppCard/>)
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<AppCard { ...mockApp }/>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('shows a title', () => {
    const { getByText } = render(<AppCard { ...mockApp }/>);
    const el = getByText('A Title');

    expect(el).toBeTruthy();
    expect(el).toBeInstanceOf(HTMLHeadingElement);
  });

  it('displays a an image for the app', () => {
    const { getByAltText } = render(<AppCard { ...mockApp }/>);
    const el = getByAltText('A Title');

    expect(el).toBeTruthy();
    expect(el).toHaveAttribute('src', 'https://image.app.com/some-image.jpg');
  });

  it('links to an app', () => {
    const { getByAltText } = render(<AppCard { ...mockApp }/>);
    const el = getByAltText('A Title').closest('a');

    expect(el).toHaveAttribute('href', 'https://location.app.com');
  });

  it('links to the repo', () => {
    const { getByText } = render(<AppCard { ...mockApp }/>);
    const el = getByText('See the code');

    expect(el).toBeTruthy();
    expect(el).toHaveAttribute('href', 'https://github.com/location/app');
  });
});
