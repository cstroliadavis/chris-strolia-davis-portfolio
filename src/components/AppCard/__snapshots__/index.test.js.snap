// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-card"
  >
    <h1
      class="title"
    >
      A Title
    </h1>
    <div
      class="app-ref"
    >
      <a
        href="https://location.app.com"
        target="_blank"
      >
        <img
          alt="A Title"
          src="https://image.app.com/some-image.jpg"
        />
      </a>
    </div>
    <div
      class="repo"
    >
      <a
        href="https://github.com/location/app"
        target="_blank"
      >
        See the code
      </a>
    </div>
    <div
      class="app-description"
    >
      <a
        href="https://location.app.com"
        target="_blank"
      >
        One of the things I worked on
      </a>
    </div>
  </div>
</DocumentFragment>
`;
